pkg/apiserver/statement/service.go

```diff
@@ -65,10 +66,19 @@ func RegisterRouter(r *gin.RouterGroup, auth *user.AuthService, s *Service) {
 			endpoint.GET("/plan/detail", s.planDetailHandler)
 
 			endpoint.POST("/download/token", s.downloadTokenHandler)
+
+			endpoint.GET("/table_columns", s.queryTableColumns)
 		}
 	}
 }
 
+func mwCacheTableColumns(c *gin.Context) {
+	if _, err := utils.GetTableColumns(statementsTable); err != nil {
+		db := utils.GetTiDBConnection(c)
+		utils.CacheTableColumns(db, statementsTable)
+	}
```

Comment:
This produce unnecessary overhead if user only access some of the API. I think a good way is to:

1. Receive `gorm.db` in cache read function.
2. Handle cache missing in the cache read function by reading from DB and cache the result.
3. Return the result from the cache.

In this way, your cache only exposes one external interface, which is `read`, while maintaining correct results in all cases, and does not have overhead when cache is not accessed.